In [83]:
import pandas as pd
sale=pd.read_excel('1 Sales Data (1).xlsx')
sale

Invoice Number       Date             Time   Zone Manager Customer  \
0              102234 2010-01-07  10:25:37.200000   East   Bruce     Polo   
1              102235 2010-01-07         12:00:00   East   Jacob      EEE   
2              102235 2010-01-07         14:19:12   East  Melvin    Marco   
3              102236 2010-01-07         12:57:00  South  Alicia      EEE   
4              102236 2010-01-07         14:51:36   East    Paul      Zee   
...               ...        ...              ...    ...     ...      ...   
26078          115273 2018-12-10         11:16:48   West  Melvin      ABC   
26079          115274 2018-12-10         11:16:48   West  Melvin      ABC   
26080          115273 2019-12-10         11:16:48   West  Melvin      ABC   
26081          115273 2019-12-10         11:16:48   West  Melvin      ABC   
26082          115274 2019-12-10         11:16:48   West  Melvin      ABC   

      Product   price  Units Sold   sales  
0         A001    485          42   20370  
1         B227    292          71   20732  
2         A009    544          87   47328  
3         D440    385          51   19635  
4          WP6    173          69   11937  
...        ...    ...         ...     ...  
26078     B229    388         202   78376  
26079     B229    388         201   77988  
26080     B229    388         500  194000  
26081     B229    388        1000  388000  
26082     B229    388        1500  582000  

[26083 rows x 10 columns]

In [84]:
print("Sum:",sale["sales"].sum()) # Find total sales

Sum: 1085806200


In [85]:
d = sale['Product '].value_counts().sum() # find total number of product 
d

26083

In [86]:
print("Maximum:",sale["sales"].max()) # Find maximum sales

Maximum: 582000


In [87]:
print("Minimum:",sale["sales"].min()) # Find minimum sales

Minimum: 865


In [88]:
print("Average:",sale["sales"].mean()) # Find the avenage sales 

Average: 41628.88471418165


In [89]:
sale.info() # null

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26083 entries, 0 to 26082
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Invoice Number  26083 non-null  int64         
 1   Date            26083 non-null  datetime64[ns]
 2   Time            26083 non-null  object        
 3   Zone            26083 non-null  object        
 4   Manager         26083 non-null  object        
 5   Customer        26083 non-null  object        
 6   Product         26083 non-null  object        
 7   price           26083 non-null  int64         
 8   Units Sold      26083 non-null  int64         
 9   sales           26083 non-null  int64         
dtypes: datetime64[ns](1), int64(4), object(5)
memory usage: 2.0+ MB


In [90]:
# find the range of sales 
Range = 582000-865
Range

581135

In [91]:
# find the standard decivation of sales
# sale.stdev.s()
#print("Standard decivation of sales:",
a=sale['sales'].std()
a

30413.734244059717

In [92]:
# find average of sales by each Zone
print("Average of sales by each Zone:",sale.groupby('Zone').mean()['sales'])



Average of sales by each Zone: Zone
East     40856.468783
North    41933.083101
South    40869.279839
West     42309.618582
Name: sales, dtype: float64


C:\Users\91984\AppData\Local\Temp\ipykernel_17880\623038399.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  print("Average of sales by each Zone:",sale.groupby('Zone').mean()['sales'])


In [93]:
# find maximum of sales by each zone
# sale("zone"),sale["sales"].max()
print("Maximum of sales by each zone:",sale.groupby('Zone').max()['sales'])


Maximum of sales by each zone: Zone
East     162656
North    165300
South    157760
West     582000
Name: sales, dtype: int64


In [94]:
# find minimum of sales by each zone
#sale("zone"),sale["sales"].max()
print("Minimum of sales by each zone:",sale.groupby('Zone').min()['sales'])


Minimum of sales by each zone: Zone
East     1015
North    1038
South     865
West      865
Name: sales, dtype: int64


# Give an inference from the data


*)This is a sales data.The data contains 26083 rows × 10 columns.
*)There is No null values in the data.
*)The total sales from 2010 to 2019 is  1085806200.
*)The Average sales from 2010 to 2019 is 41628.88471418165.
*)The Maximum sales from 2010 to 2019 is  582000.
*)The Minimum sales from 2010 to 2019 is  865
*)The range of sales is 581135. 
*)The Product sales from 2010 to 2019 is 26083.
*)The standard decivation of sales from 2010 t0 2019 is 30413.734244059717

In [95]:
#Find Manager Wise sales.
print("Find Manager Wise sales:",sale.groupby("Manager").sum()["sales"])

Find Manager Wise sales: Manager
Aaron         26849758
Abbott        27098975
Abel          28249474
Alexandra     25981767
Alice         26573341
Alicia        25575993
Allan         90291223
Bruce         25533890
Grace         25789066
Hannah        27736138
Jack          27608183
Jacob         27505184
Jim          267417319
Lucy          24248866
Lydia         24951596
Malcolm       24938730
Melvin        29764672
Paul          28634113
Paxton        26878183
Rojer        116239848
Sarah         23662807
Tom          109184168
Tony          25092906
Name: sales, dtype: int64


C:\Users\91984\AppData\Local\Temp\ipykernel_17880\1121293573.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  print("Find Manager Wise sales:",sale.groupby("Manager").sum()["sales"])


In [96]:
#Find product wise sales.
print("Find product wise sales:",sale.groupby("Product ").sum()["sales"])

Find product wise sales: Product 
A001    71165990
A002    83824255
A003    95344384
A009    80055584
A010    36526140
AL10    81071385
AL9     63099648
B227    43665096
B228    78313572
B229    59977816
D440    55816915
D441    50352855
G224    52050360
G225    73537764
KKK     60218836
WP6     25954152
WP7     74831448
Name: sales, dtype: int64


C:\Users\91984\AppData\Local\Temp\ipykernel_17880\1875215446.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  print("Find product wise sales:",sale.groupby("Product ").sum()["sales"])


In [97]:
#Find zone Wise Sales.
print("Find zone Wise Sales:",sale.groupby("Zone").sum()["sales"])

Find zone Wise Sales: Zone
East     307567497
North    361295444
South    101355814
West     315587445
Name: sales, dtype: int64


C:\Users\91984\AppData\Local\Temp\ipykernel_17880\2859583499.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  print("Find zone Wise Sales:",sale.groupby("Zone").sum()["sales"])


In [98]:
#Find zone wise product sales.
print("#Find zone wise product sales:",sale.groupby(["Zone","Product "]).sum()["sales"])

#Find zone wise product sales: Zone  Product 
East  A001        17583190
      A002        26627745
      A003        23718990
      A009        24735680
      A010        11311020
                    ...   
West  G224        14399892
      G225        22970988
      KKK         16949557
      WP6          6044793
      WP7         24498808
Name: sales, Length: 68, dtype: int64


C:\Users\91984\AppData\Local\Temp\ipykernel_17880\2075895004.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  print("#Find zone wise product sales:",sale.groupby(["Zone","Product "]).sum()["sales"])


In [99]:
#Find Average units sold by each manager.
print("Find Average units sold by each manager:",sale.groupby("Manager").mean()["Units Sold"])

Find Average units sold by each manager: Manager
Aaron        115.607595
Abbott       113.591772
Abel         117.488924
Alexandra    115.731013
Alice        111.643987
Alicia       114.828704
Allan        112.941204
Bruce        110.144817
Grace        115.830696
Hannah       116.110759
Jack         113.547468
Jacob        113.286585
Jim          112.900398
Lucy         110.412975
Lydia        115.243671
Malcolm      109.780063
Melvin       118.268589
Paul         119.487342
Paxton       111.237342
Rojer        112.920000
Sarah        110.454114
Tom          115.477492
Tony         108.530063
Name: Units Sold, dtype: float64


C:\Users\91984\AppData\Local\Temp\ipykernel_17880\2293601089.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  print("Find Average units sold by each manager:",sale.groupby("Manager").mean()["Units Sold"])


In [100]:
#Find average units sold by zone.
print("Find average units sold by zone:",sale.groupby("Zone").mean()["Units Sold"])

Find average units sold by zone: Zone
East     112.711743
North    113.574048
South    112.649597
West     114.656522
Name: Units Sold, dtype: float64


C:\Users\91984\AppData\Local\Temp\ipykernel_17880\3418028132.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  print("Find average units sold by zone:",sale.groupby("Zone").mean()["Units Sold"])


In [101]:
#Find total sales by customer.
print("Find total sales by customer:",sale.groupby("Customer").sum()["sales"])

Find total sales by customer: Customer
ABC      157258814
AX       205881436
EEE      204380240
Marco    258936887
Polo     153491179
Zee      105857644
Name: sales, dtype: int64


C:\Users\91984\AppData\Local\Temp\ipykernel_17880\1805134934.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  print("Find total sales by customer:",sale.groupby("Customer").sum()["sales"])


In [102]:
#Find total units sold by manager.
print("Find total units sold by manager:",sale.groupby("Manager").sum()["Units Sold"])

Find total units sold by manager: Manager
Aaron         73064
Abbott        71790
Abel          74253
Alexandra     73142
Alice         70559
Alicia        74409
Allan        243953
Bruce         72255
Grace         73205
Hannah        73382
Jack          71762
Jacob         74316
Jim          737917
Lucy          69781
Lydia         72834
Malcolm       69381
Melvin        77939
Paul          75516
Paxton        70302
Rojer        316176
Sarah         69807
Tom          287308
Tony          68591
Name: Units Sold, dtype: int64


C:\Users\91984\AppData\Local\Temp\ipykernel_17880\733586631.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  print("Find total units sold by manager:",sale.groupby("Manager").sum()["Units Sold"])


In [107]:
#Find Total Number of Orders
print("Find Total Number of Orders:",sale['Invoice Number'].count().sum())

Find Total Number of Orders: 26083


In [109]:
print("Find Total Number of Orders:",sale['Invoice Number'].count())

Find Total Number of Orders: 26083


Give an inference from the data

*)This is a sales data.The data contains 26083 rows × 10 columns.
*)There is No null values in the data.
*)The Total Number of Orders from 2010 to 2019 is 26083.